tests/test_widget.py

```diff
@@ -157,3 +156,29 @@ def test_widget_mount_ids_must_be_unique_mounting_multiple_calls(parent):
     parent.mount(widget1)
     with pytest.raises(DuplicateIds):
         parent.mount(widget2)
+
+
+# Regression test for https://github.com/Textualize/textual/issues/1634
+async def test_remove():
+    class RemoveMeLabel(Label):
+        async def on_mount(self) -> None:
+            await self.action("app.remove_all")
+
+    class Container(Widget):
+        async def clear(self) -> None:
+            await self.query("*").remove()
+
+    class RemoveApp(App):
+        def compose(self) -> ComposeResult:
+            yield Container(RemoveMeLabel())
+
+        async def action_remove_all(self) -> None:
+            await self.query_one(Container).clear()
+            self.exit(123)
+
+    app = RemoveApp()
+    async with app.run_test() as pilot:
+        await pilot.press("r")
+        await pilot.pause()
+    assert app.return_value == 123
+    assert True
```

Comment:
```suggestion
```

Reply:
I mean, it's not wrong 😆

Reply:
It's a very common cosmic ray detection system. If it fails you know a cosmic ray was detected.

Reply:
My bad. A placeholder while I was finding something to assert on.